src/app/appRouter.tsx

```diff
@@ -1,15 +1,34 @@
-import { createBrowserRouter } from 'react-router-dom'
+import { type ReactNode } from 'react'
+import { Navigate, createBrowserRouter } from 'react-router-dom'
 import { featureToggleLoader } from '@/entities/featureToggle'
+import { selectIsAuthorized } from '@/entities/session'
 import { CartPage } from '@/pages/cart'
 import { CategoryPage } from '@/pages/category'
 import { LoginPage } from '@/pages/login'
 import { MainPage } from '@/pages/main'
 import { ProductPage } from '@/pages/product'
 import { WishlistPage } from '@/pages/wishlist'
+import { useAppSelector } from '@/shared/model'
 import { appStore } from './appStore'
 import { baseLayout } from './layouts/baseLayout'
 import { layoutWithSidebar } from './layouts/layoutWithSidebar'
 
+const GuestGuard = ({ children }: { children: ReactNode }) => {
```

Comment:
React components are written with function declaration in the project. Let's rewrite this:

```suggestion
type GuardProps = {...}

function GuestGuard({ children }: GuardProps) {
```